### PR TITLE
changes for updating to megatron brnach fairseq_v3

### PR DESCRIPTION
**Pull request by ngoyal2707**
updated to megatron_v3 without seq_parallel for now.

will send a separate PR for that once this lands

### CODE REVIEW

File: metaseq/models/transformer_decoder.py [hunk 1]
```diff
@@ -473,7 +474,7 @@ def max_positions(self):
         return min(self.max_target_positions, self.embed_positions.max_positions)
 
     def buffered_future_mask(self, tensor, input_tokens=None):
-        batch_size, cur_seq_len = tensor.size(0), tensor.size(1)
+        cur_seq_len, batch_size = tensor.size(0), tensor.size(1)
```
Comment (suchenzang) [inline metaseq/models/transformer_decoder.py:477]: Oh noes.. lost me here - how did this flip?

**Comment by suchenzang** [comment]
Nits on lint & not sure why GPU test is now hanging...

**Review by suchenzang — APPROVED** [review]
Stamping to unblock for subsequent PR

**Comment by punitkoura** [comment]
> Nits on lint & not sure why GPU test is now hanging...

I think they are failing somewhere... (The tests hang if something internally fails since we fork processes for distributed model loading). @ngoyal2707 could you run the tests locally once? This would indicate we're not able to load and prompt the 125m model.

**Comment by ngoyal2707** [comment]
hmm running locally I get following errors, none of these seems related to this PR:

```
FAILED gpu_tests/test_hf_compatibility.py::TestHFCompatibility::test_model_parallel_metaseq_hf_compatibility - Exception: Error while reading vocab & merges files: No such file or directory (os error 2)
FAILED gpu_tests/test_hf_compatibility.py::TestHFCompatibility::test_singleton_metaseq_hf_compatibility - Exception: Error while reading vocab & merges files: No such file or directory (os error 2)
FAILED tests/test_lm_context_window.py::TestLMContextWindow::test_eval_dataloader - TypeError: __init__() takes 2 positional arguments but 3 were given
FAILED tests/test_memory_efficient_fp16.py::TestMemoryEfficientFP16::test_load_state_dict - omegaconf.errors.ConfigAttributeError: Missing key fp16_adam_stats
FAILED tests/test_polynomial_lr.py::TestPolynomialLRScheduler::test_polynomial_decay_no_adam_warmup - omegaconf.errors.ConfigAttributeError: Missing key fp16_adam_stats
FAILED tests/test_polynomial_lr.py::TestPolynomialLRScheduler::test_polynomial_decay_with_adam_warmup - omegaconf.errors.ConfigAttributeError: Missing key fp16_adam_stats
FAILED tests/test_sequence_generator.py::TestJitSequenceGenerator::test_export_transformer - AttributeError: 'Namespace' object has no attribute 'decoder_input_dim'
FAILED tests/test_sequence_generator.py::TestJitSequenceGenerator::test_sequence_generator - AttributeError: 'Namespace' object has no attribute 'decoder_input_dim'
FAILED tests/test_sequence_generator.py::TestSequenceGenerator::test_maxlen - RuntimeError: The expanded size of the tensor (2) must match the existing size (3) at non-singleton dimension 1.  Target sizes: [4, 2].  Tensor sizes: [4, 3]
FAILED tests/test_sequence_generator.py::TestSequenceGenerator::test_with_normalization - TypeError: forward() got an unexpected keyword argument 'incremental_state'
FAILED tests/test_sequence_generator.py::TestSequenceGenerator::test_without_normalization - TypeError: forward() got an unexpected keyword argument 'incremental_state'
FAILED tests/test_streaming_language_modeling_task.py::TestReproducibility::test_mid_epoch_reproducibility - SystemExit: 2
FAILED tests/test_streaming_language_modeling_task.py::TestReproducibility::test_reproducibility - SystemExit: 2
FAILED tests/test_streaming_language_modeling_task.py::TestReproducibility::test_reproducibility_fp16 - SystemExit: 2
FAILED tests/test_streaming_language_modeling_task.py::TestReproducibility::test_reproducibility_memory_efficient_fp16 - SystemExit: 2
FAILED tests/distributed/test_utils.py::TestBroadcastObject::test_complex - torch.multiprocessing.spawn.ProcessRaisedException:
ERROR gpu_tests/test_hub_utils.py::test_generator_interface
ERROR gpu_tests/test_hub_utils.py::test_filter_special
```

**Comment by punitkoura** [comment]
> hmm running locally I get following errors, none of these seems related to this PR:
> 
> ```
> FAILED gpu_tests/test_hf_compatibility.py::TestHFCompatibility::test_model_parallel_metaseq_hf_compatibility - Exception: Error while reading vocab & merges files: No such file or directory (os error 2)
> FAILED gpu_tests/test_hf_compatibility.py::TestHFCompatibility::test_singleton_metaseq_hf_compatibility - Exception: Error while reading vocab & merges files: No such file or directory (os error 2)
> FAILED tests/test_lm_context_window.py::TestLMContextWindow::test_eval_dataloader - TypeError: __init__() takes 2 positional arguments but 3 were given
> FAILED tests/test_memory_efficient_fp16.py::TestMemoryEfficientFP16::test_load_state_dict - omegaconf.errors.ConfigAttributeError: Missing key fp16_adam_stats
> FAILED tests/test_polynomial_lr.py::TestPolynomialLRScheduler::test_polynomial_decay_no_adam_warmup - omegaconf.errors.ConfigAttributeError: Missing key fp16_adam_stats
> FAILED tests/test_polynomial_lr.py::TestPolynomialLRScheduler::test_polynomial_decay_with_adam_warmup - omegaconf.errors.ConfigAttributeError: Missing key fp16_adam_stats
> FAILED tests/test_sequence_generator.py::TestJitSequenceGenerator::test_export_transformer - AttributeError: 'Namespace' object has no attribute 'decoder_input_dim'
> FAILED tests/test_sequence_generator.py::TestJitSequenceGenerator::test_sequence_generator - AttributeError: 'Namespace' object has no attribute 'decoder_input_dim'
> FAILED tests/test_sequence_generator.py::TestSequenceGenerator::test_maxlen - RuntimeError: The expanded size of the tensor (2) must match the existing size (3) at non-singleton dimension 1.  Target sizes: [4, 2].  Tensor sizes: [4, 3]
> FAILED tests/test_sequence_generator.py::TestSequenceGenerator::test_with_normalization - TypeError: forward() got an unexpected keyword argument 'incremental_state'
> FAILED tests/test_sequence_generator.py::TestSequenceGenerator::test_without_normalization - TypeError: forward() got an unexpected keyword argument 'incremental_state'
> FAILED tests/test_streaming_language_modeling_task.py::TestReproducibility::test_mid_epoch_reproducibility - SystemExit: 2
> FAILED tests/test_streaming_language_modeling_task.py::TestReproducibility::test_reproducibility - SystemExit: 2
> FAILED tests/test_streaming_language_modeling_task.py::TestReproducibility::test_reproducibility_fp16 - SystemExit: 2
> FAILED tests/test_streaming_language_modeling_task.py::TestReproducibility::test_reproducibility_memory_efficient_fp16 - SystemExit: 2
> FAILED tests/distributed/test_utils.py::TestBroadcastObject::test_complex - torch.multiprocessing.spawn.ProcessRaisedException:
> ERROR gpu_tests/test_hub_utils.py::test_generator_interface
> ERROR gpu_tests/test_hub_utils.py::test_filter_special
> ```

@ngoyal2707 Sorry, you would need to do these prerequisite steps to be able to run the tests... https://github.com/facebookresearch/metaseq/blob/main/.circleci/config.yml#L130

**Comment by punitkoura** [comment]
Also, you can only try running this test first gpu_tests/test_checkpoint_saving.py  since the tests seem to hang here

File: metaseq/models/transformer_decoder.py (hunk 1 above)
Comment (ngoyal2707) [reply metaseq/models/transformer_decoder.py:477]: ohh cause, I am not converting to T x B x C before this call

**Pull request merged**